neonvm/controllers/virtualmachine_controller.go

```diff
@@ -503,9 +503,68 @@ func (r *VirtualMachineReconciler) doReconcile(ctx context.Context, virtualmachi
 		// TODO: implement RestartPolicy
 		switch virtualmachine.Spec.RestartPolicy {
 		case vmv1.RestartPolicyAlways:
-			// TODO: restart VirtualMachine
+			log.Info("Restarting VM runner pod", "VM.Phase", virtualmachine.Status.Phase, "RestartPolicy", virtualmachine.Spec.RestartPolicy)
+			// get runner to delete
+			vmRunner := &corev1.Pod{}
+			err := r.Get(ctx, types.NamespacedName{Name: virtualmachine.Status.PodName, Namespace: virtualmachine.Namespace}, vmRunner)
+			if err == nil {
+				// delete current runner
+				if err = r.Delete(ctx, vmRunner); err != nil {
+					log.Error(err, "Failed to delete runner Pod", "Pod.Namespace", vmRunner.Namespace, "Pod.Name", vmRunner.Name)
```

Comment:
my instinct is that we should avoid logging errors and then returning them: if you do this every time you get an error, you end up with one error being logged potentially many times. I don't, from this diff, how much logging we end up doing, but it seems like it could be a lot.